public/app/features/alerting/unified/components/rules/deleted-rules/DeletedRules.tsx

```diff
@@ -18,7 +20,8 @@ interface DeletedRulesProps {
 export function DeletedRules({ deletedRules }: DeletedRulesProps) {
   const [confirmRestore, setConfirmRestore] = useState(false);
   const [restoreRule, setRestoreRule] = useState<RulerGrafanaRuleDTO | undefined>();
-
+  const [confirmDeletePermanently, setConfirmDeletePermanently] = useState(false);
+  const [guidToDelete, setGuidToDelete] = useState<string | undefined>();
```

Comment:
Do we need both variables?
Maybe we could control displaying the modal by using `guidToDelete`
For example:
```
const confirmDeletePermanently = guidToDelete === undefined
```

Reply:
you're right!